x-pack/plugins/observability_solution/metrics_data_access/server/routes/metrics_explorer/index.ts

```diff
@@ -24,61 +21,79 @@ import { queryTotalGroupings } from './lib/query_total_groupings';
 import { transformSeries } from './lib/transform_series';
 import { KibanaFramework } from '../../lib/adapters/framework/kibana_framework_adapter';
 
-const escapeHatch = schema.object({}, { unknowns: 'allow' });
-
 export const initMetricExplorerRoute = (framework: KibanaFramework) => {
+  const validateBody = createRouteValidationFunction(metricsExplorerRequestBodyRT);
   framework.registerRoute(
     {
       method: 'post',
       path: '/api/infra/metrics_explorer',
       validate: {
-        body: escapeHatch,
+        body: validateBody,
       },
     },
     async (requestContext, request, response) => {
-      const options = pipe(
-        metricsExplorerRequestBodyRT.decode(request.body),
-        fold(throwErrors(Boom.badRequest), identity)
-      );
+      const options = request.body;
 
-      const client = createSearchClient(requestContext, framework);
-      const interval = await findIntervalForMetrics(client, options);
+      try {
+        if (options.metrics.length > METRICS_EXPLORER_API_MAX_METRICS) {
```

Comment:
Could this endpoint potentially be used in alerts or other scenarios where a user might exceed the 20-metric limit and start getting this error?

Reply:
You mean our internal alerts? This endpoint was used to render the Metrics Threshold rule charts, but it has been replaced by Lens in this PR https://github.com/elastic/kibana/pull/184950. So it's no longer used anywhere else but the Metrics Explorer page.